src/clr-angular/data/datagrid/datagrid-row.html

```diff
@@ -20,7 +20,11 @@
     depending on whether the details replace the row or not.
 -->
 <ng-template #detail>
-  <ng-content select="clr-dg-row-detail"></ng-content>
+  <div [id]="expandableId">
```

Comment:
Why is this id not on the `clr-dg-row-detail`? We are creating an additional wrapper div just to add this id. Plus, we should try to isolate these expanded row changes within its own component.

Reply:
It's a bit of unwanted hack to make the `aria-control` work. The id is for `[attr.aria-controls]="expandableId"` few lines below. 

`aria-controls` require to have ID to witch to link itself. So this is the only way I found to link them both. If `clr-dg-row-detail` create it own ID, it will have to pass it to the parent. If we create ID from the parent we will have to find a way to pass it below. Or we will have to create service to track this. All the solutions are a bit overkill for something that is not even supported on more than one SR. 

If we decide to drop the `aria-control` that I still think is not well supported and is adding unneeded changes we could drop the `<div id="expandableId"></div>` and the additional attribute to the button below.

Reply:
I understand you need `aria-control` as well as an ID for that. What I am saying is that we are creating additional wrappers. And really this ID belongs to the row detail. What you need is uniq id provider that is shared between datagrid row and its row detail components.

Reply:
I don't see a reason to have a NEW provider just to handle this case. `aria-control` is not a well-supported attribute and it's not worth the additional complexity of a provider. 

And when it comes to providing and accessing the ID is all done in one place. It's clearly visible when the id is pointing.

Reply:
But you are complicating the template structure by adding more wrappers just to place that one attribute on them. And that's not even added in `clr-dg-row-detail`'s own template. Keep these changes where they belong to.

I don't see how injecting unique id provider would introduce more complexity. You could also just add a unique id property to some service that both `clr-dg-row` and `clr-dg-row-detail` share.

Reply:
Can you provide some code examples on how this has to be done? I don't see a common service between both components?

The changes are located inside the template for `clr-dg-row-details` so when and if this is removed they will go away as well.

But if you could provide a better solution to that I'm ok with it.